src/utils/connectionStringParser.ts

```diff
@@ -0,0 +1,47 @@
+import { ConnectionString, ConnectionStringKey } from '../Declarations/Contracts';
+
+import * as Constants from '../Declarations/Constants';
+
+export class ConnectionStringParser {
+  private static readonly FIELDS_SEPARATOR = ';';
+
+  private static readonly FIELD_KEY_VALUE_SEPARATOR = '=';
+
+  public static parse(connectionString?: string): ConnectionString {
+    if (!connectionString) {
+      return {};
+    }
+
+    const kvPairs = connectionString.split(ConnectionStringParser.FIELDS_SEPARATOR);
+
+    const result: ConnectionString = kvPairs.reduce((fields: ConnectionString, kv: string) => {
+      const kvParts = kv.split(ConnectionStringParser.FIELD_KEY_VALUE_SEPARATOR);
+
+      if (kvParts.length === 2) {
```

Comment:
I believe if split and kvParts is not actually a pair then the connection string is invalid (should fail).

Reply:
Just to clarify, do you mean the entire connection string is invalid? Or just the particular kv pair?

Reply:
Both. `foo=bar=test` would split into a kvParts of length 3, so it is not a pair and invalid, and therefore, makes the entire connection string invalid.

Reply:
Fixed ptal